ethclient/ethclient.go

```diff
@@ -163,7 +163,7 @@ func (ec *Client) TransactionByHash(ctx context.Context, hash common.Hash) (tx *
 	} else if len(raw) == 0 {
 		return nil, false, ethereum.NotFound
 	}
-	if err := json.Unmarshal(raw, tx); err != nil {
+	if err := json.Unmarshal(raw, &tx); err != nil {
```

Comment:
Wouldn't a cleaner solution actually be adding `tx = new(types.Transaction)` before the original line? the double pointer seems a bit misleading to me, but if you think it's fine, then LGTM to me too.